### ISSUE TITLE
Inline disabling of Style/NumericLiterals is ignored when generating MaxLength in the todo file

### ISSUE THREAD

**Issue by reedstonefood** [rubocop/rubocop#10066]
Using `# rubocop:disable Style/NumericLiterals` inline, works fine when running `rubocop`.

However, the length of the disabled literal is used by the logic that creates `.rubocop_todo.yml` file. This is a problem when the literal that has been excluded for the cop is longer than the existing value of `MinDigits` in the `todo` file. This means that the `MinDigits` parameter can get set to something larger than is desired, which eventually means it's impossible to use `# rubocop:disable Style/NumericLiterals` on numbers that are longer than `MinDigits` as I go on to describe.

--------

## Expected behavior

Running `rubocop --regenerate-todo` should not change the `MaxLength` value.

## Actual behavior

Running `rubocop --regenerate-todo` increases the value of `MinDigits` to be one more than the length of the literal that we have instructed the `Style/NumericLiterals` to ignore.

This then leads to follow-up problems - the next run of `rubocop` will not be green due to an un-necessary `disable` comment being detected, and removing the `disable` comment is undesirable as we intended to mark that specific literal is being OK.

## Steps to reproduce the problem

1. Create a 10 digit constant somewhere `FIRST_VIOLATION = 1234567890`
2. Run `rubocop --regenerate-todo` which should generate something like
```
# Offense count: 1
# Cop supports --auto-correct.
# Configuration parameters: Strict.
Style/NumericLiterals:
  MinDigits: 11
```
3. Create a 20 digit constant somewhere, but add an inline disable.
```ruby
SECOND_VIOLATION = 12345678901234567890`
```
4. Run `rubocop` and everything is green. It's all good and as desired up to this point.
5. Now run `rubocop --regenerate-todo` again. It will get updated to
```
# Offense count: 2
# Cop supports --auto-correct.
# Configuration parameters: Strict.
Style/NumericLiterals:
  MinDigits: 21
```
6. Run `rubocop` and it doesn't run succesfully - it tells you the `disable` is no longer required.
7. Remove the inline `disable` and `rubocop` will return green. However, now the special case which we want to allow, is no longer explicitly allowed and is now part of the `todo` file.

## RuboCop version

```
1.18.4 (using Parser 3.0.2.0, rubocop-ast 1.8.0, running on ruby 2.7.1 x86_64-darwin19)
  - rubocop-rails 2.11.3
```

**Comment by reedstonefood** [comment]
The workaround I'm using for this is for now to `Exclude` the whole file in `rubocop.yml` - this is respected by the `todo` generation code. So it looks like this problem is specific to when a cop is disabled inline.
